### PR TITLE
Resolve mkdocs issues

### PR DESCRIPTION
**Pull request by LinasKo**
# Description

Addressed all issues highlighted when running `mkdocs serve`.

## Type of change

Please delete options that are not relevant.

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

1. Ran `mkdocs serve` making sure there are no issues
2. Looked through the docs, verifying there are no regressions

## Any specific deployment considerations

Extra dependency: We're now explicitly requiring a version of `mkdocstrings-python`, as that was not handled by `mkdocstrings[python]`.

## Docs

-   [x] Docs updated? What were the changes:

Added whitespaces and `Any` types, primarily.

### CODE REVIEW

**Comment by onuralpszr** [comment]
@LinasKo 

- I added code-spell check, ignore notebooks (read as txt not notebook and that cause so many errors) 
- I added mdformat/ruff/mkdocs to make complete re-formatting

**Review by onuralpszr — APPROVED** [review]

**Comment by LinasKo** [comment]
I'll give it another look as well

File: README.md [hunk 1]
```diff
@@ -102,10 +102,7 @@ image = cv2.imread(...)
 detections = sv.Detections(...)
 
 box_annotator = sv.BoxAnnotator()
-annotated_frame = box_annotator.annotate(
-    scene=image.copy(),
-    detections=detections
-)
+annotated_frame = box_annotator.annotate(scene=image.copy(), detections=detections)
```
Comment (LinasKo) [inline README.md:105]: This is less readable

File: README.md [hunk 2]
```diff
@@ -138,88 +135,69 @@ for path, image, annotation in ds:
...
 
-  ```python
-  dataset = sv.DetectionDataset.from_yolo(
-      images_directory_path=...,
-      annotations_directory_path=...,
-      data_yaml_path=...
-  )
+    ```python
+    dataset = sv.DetectionDataset.from_yolo(
+        images_directory_path=..., annotations_directory_path=..., data_yaml_path=...
```
Comment (LinasKo) [inline README.md:140]: this is much less readable

File: README.md [hunk 3]
```diff
@@ -138,88 +135,69 @@ for path, image, annotation in ds:
...
-  ).as_pascal_voc(
-      images_directory_path=...,
-      annotations_directory_path=...
-  )
-  ```
+    ```python
+    sv.DetectionDataset.from_yolo(
+        images_directory_path=..., annotations_directory_path=..., data_yaml_path=...
+    ).as_pascal_voc(images_directory_path=..., annotations_directory_path=...)
+    ```
```
Comment (LinasKo) [inline README.md:200]: This won't work for as as the formatter, if we can't make it more lenient.

**Comment by LinasKo** [comment]
Hey @onuralpszr, I glanced at the results - I'm really not fond of the situation where it reformats arguments split across multiple lines into a format where it's all on a single one (or packed into as few lines as possible).

This is less pleasant to look at than what we had, and we have many many cases like this.

Let's either find a way to fix it, or split the markdown formatter into a separate PR.

**Comment by onuralpszr** [comment]
> Hey @onuralpszr, I glanced at the results - I'm really not fond of the situation where it reformats arguments split across multiple lines into a format where it's all on a single one (or packed into as few lines as possible).
> 
> This is less pleasant to look at than what we had, and we have many many cases like this.
> 
> Let's either find a way to fix it, or split the markdown formatter into a separate PR.

Let me take look quickly to handle this (I got something in my mind)

**Comment by onuralpszr** [comment]
@LinasKo I revert code parts and re-adjusted them, code-spell is our new tool for catch spelling errors and typos. I am merging in. Thank you.

**Comment by LinasKo** [comment]
Excellent work - I can't see any regressions.

**Pull request merged**